frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +86,28 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // saveDocumentBeforeCheck needed in order to update the state of the .git/machete VirtualFile before actual check
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> saveDocumentBeforeCheck(file));
+    try {
+      val branchNamesFromFile = VfsUtil.loadText(file.getVirtualFile());
+      if (branchNamesFromFile.indexOf(processedBranchName) != branchNamesFromFile.lastIndexOf(processedBranchName)) {
+        holder.newAnnotation(HighlightSeverity.ERROR,
+            getNonHtmlString("string.GitMachete.MacheteAnnotator.branch-entry-already-defined")
+                .format(processedBranchName))
+            .range(TextRange.from(branchNamesFromFile.lastIndexOf(processedBranchName), processedBranchName.length()))
+            .create();
+      }
+    } catch (PluginException | IllegalStateException ignored) { // ignore dubious IDE checks against annotation range
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @UIEffect
+  private void saveDocumentBeforeCheck(PsiFile file) {
+    val fileDocManager = FileDocumentManager.getInstance();
+    fileDocManager.saveDocument(Objects.requireNonNull(fileDocManager.getDocument(file.getVirtualFile())));
```

Comment:
Hmm if `fileDocManager.getDocument(...)` is null, then a runtime exception is thrown right? 🤔 do we have a guarantee that it'll indeed always be non-null? I'd add some `if (... != null)` if not

Reply:
In my opinion, we do should not add additional check. Reasoning:
1. This method is called from within `private void processMacheteGeneratedEntry` method
2. The `private void processMacheteGeneratedEntry` method creates a PsiFile object earlier as follows:
`PsiFile file = macheteEntry.getContainingFile();`
3. The `PsiFile file` parameter used in this method is the same `file` object
4. Therefore, on this stage of the `processMacheteGeneratedEntry` logic, `file` object should not be null. If it would be null, then an exception would be thrown earlier, for example here:
`val branchNames = file.getBranchNamesForPsiMacheteFile();`
5. `file` should not be null, so `fileDocManager.getDocument(...)` should not be null too.
6. If for some reason `file` would be in fact a `null` at the moment this helper method is called - then a `NullPointerException` will occur. But this should be good in my opinion. We should get an exception, if something has been broken unexpectedly with the file.

What do you think?

Reply:
Okay! let's leave as is then